python/img_hdf5_reader.py

```diff
@@ -235,36 +135,90 @@ def _get_shape_and_dtype(self, channels):
+    json_file, url, name, channel_names, is_pyramid=False, transform=None
+):
+    if not transform:
+        transform = {"translate": {"y": 0, "x": 0}, "scale": 1}
+    raster_json = {
+        "version": "0.0.1",
+        "images": [
+            {
+                "name": name,
+                "url": url,
+                "type": "zarr",
+                "metadata": {
+                    "dimensions": create_dimensions(channel_names),
+                    "is_pyramid": is_pyramid,
+                    "transform": transform,
+                },
+            }
+        ],
+    }
+    json.dump(raster_json, json_file, indent=2)
+
+
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(
         description='Create PNG files and one JSON file with metadata')
     parser.add_argument(
         '--hdf5', required=True,
         help='HDF5 file with raster data')
-    parser.add_argument(
-        '--json_file', required=True, type=argparse.FileType('x'),
-        help='JSON file which will include image dimensions and location')
     parser.add_argument(
         '--zarr_file', required=True,
         help='Directory to write zarr output to.')
     parser.add_argument(
         '--channels', required=True,
         help='List of channels to include in zarr image.')
     parser.add_argument(
-        '--tiles_url', required=True,
+        '--server_url', required=True,
         help='Output URL to zarr store.')
     parser.add_argument(
         '--sample', default=1, type=int,
         help='Sample 1 pixel out of N')
+    parser.add_argument(
+        '--tile_size', default=DEFAULT_TILE_SIZE,
+        type=int, help='Size of zarr image tiles.'
+    )
+    parser.add_argument(
+        '--raster_json', required=True, type=argparse.FileType('x'),
+        help='JSON file which will include image dimensions and location')
+    parser.add_argument(
+        "--raster_name", required=True, help="Image name for metadata.",
+    )
+    parser.add_argument(
+        "--as_pyramid", default=True, help="Whether to generate image pyramid."
+    )
     args = parser.parse_args()
 
     reader = ImgHdf5Reader(args.hdf5)
 
+    zarr_path = Path(args.zarr_file)
     channels = [c for c in args.channels.split(',')]
+    is_pyramid = args.as_pyramid
+    destination_url = urllib.parse.urljoin(
+        args.server_url, zarr_path.name
+    )
+
     reader.to_zarr(
+        output_path=zarr_path,
         channels=channels,
         sample=args.sample,
-        json_file=args.json_file,
-        tiles_url=args.tiles_url,
-        zarr_file=args.zarr_file,
+        tile_size=args.tile_size,
+        is_pyramid_base=is_pyramid,
+    )
+
+    if is_pyramid:
+        tile_zarr(str(zarr_path / "0"))
+        # Consolidate metadata into single .zmetadata for pyramid
+        # https://zarr.readthedocs.io/en/stable/tutorial.html#consolidating-metadata
+        z_group = zarr.open(str(zarr_path))
+        zarr.consolidate_metadata(z_group.store)
```

Comment:
This isn't actually going to be used, right?  Like, `zarr.js` will still use the `.zarray` file for parsing?  If this is only for future use, maybe a comment about that would be helpful